src/firefly/js/charts/ui/PlotlyToolbar.jsx

```diff
@@ -44,6 +45,38 @@ export class ScatterToolbar extends SimpleComponent {
     }
 }
 
+function isSelectable(tbl_id, chartId, type) {
```

Comment:
I don't quite understand how we decide that selection box should be present. We might be creating unnecessary complexity here. What if x and y are expressions? I think we should not allow selection box for "unknown chart types".

Reply:
Yes, we need to determine 
- if showing the selection box for  'unknown chart types'? 
   I had asked the original design intention if showing XY options in case x, y data is defined for 'unknown chart types', and the reply is 'the intention was to exclude XY options from the charts that do not have XY axes', so should XY options be excluded if the chart which is 'unknown type' but with x, y data?
- If not excluded, selection box is shown in case XY is defined and mapped to table columns directly (per discussion with Trey). Selection box could be rendered if x, y is expressions later once we add new columns into the table for the expression. 

This is thought that makes the code so far.

Reply:
The problem is that firefly enhanced charts might be creating the x and y arrays using custom logic bypassing mappings. May be we can check fireflyData.type here: for fireflyHeatmap we do want to support area selection?

Also you can use getColValidator in ColumnOrExpression.jsx to check for column expressions. TableStatsCntlr.getColValStats(tblId) will return table statistics.

Reply:
Please check ffapi-slate-test2.html, the selection area is not changed for heatmap and scatter cases. (the firefly chart type implemented is not changed). In test2 example, originally scatter chart shows no selection box, and heatmap charts shows the selection box.

Reply:
The area selection issue for  heatmap in ffapi-highlevel-charttest.html is solved by add table columns info into the state for PlotlyToolBar for updating the tool bar when the table loading is complete.